src/pages/home/report/withReportAndReportActionOrNotFound.js

```diff
@@ -0,0 +1,147 @@
+        // So we need to call openReport here for small screen
+        const firstRef = useRef(true);
+        useEffect(() => {
+            if (!firstRef.current) {
+                return;
+            }
+            firstRef.current = false;
+            if (props.isSmallScreenWidth) {
+                Report.openReport(props.route.params.reportID);
+            }
+        }, [props.isSmallScreenWidth, props.route.params.reportID]);
+
+        const isLoadingReport = props.isLoadingReportData && (_.isEmpty(props.report) || !props.report.reportID);
+        const shouldHideReport = !isLoadingReport && (_.isEmpty(props.report) || !props.report.reportID || !ReportUtils.canAccessReport(props.report, props.policies, props.betas));
+
+        let reportAction = props.reportActions[`${props.route.params.reportActionID.toString()}`];
+        // Handle threads if needed
+        if (reportAction === undefined || reportAction.reportActionID === undefined) {
+            reportAction = ReportActionsUtils.getParentReportAction(props.report);
+        }
+        const isLoadingReportAction = _.isEmpty(props.reportActions) || (props.report.isLoadingReportActions && _.isEmpty(reportAction));
+
+        if ((isLoadingReport || isLoadingReportAction) && !shouldHideReport) {
+            return <FullscreenLoadingIndicator />;
+        }
+        if (shouldHideReport || _.isEmpty(reportAction)) {
+            return <NotFoundPage />;
+        }
+        const rest = _.omit(props, ['forwardedRef']);
+        return (
+            <WrappedComponent
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...rest}
+                ref={props.forwardedRef}
+            />
+        );
+    }
+
+    WithReportAndReportActionOrNotFound.propTypes = propTypes;
+    WithReportAndReportActionOrNotFound.defaultProps = defaultProps;
+    WithReportAndReportActionOrNotFound.displayName = `withReportAndReportActionOrNotFound(${getComponentDisplayName(WrappedComponent)})`;
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    const withReportAndReportActionOrNotFound = React.forwardRef((props, ref) => (
+        <WithReportAndReportActionOrNotFound
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            forwardedRef={ref}
+        />
+    ));
+
+    return compose(
+        withWindowDimensions,
+        withOnyx({
+            report: {
+                key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT}${route.params.reportID}`,
+            },
+            isLoadingReportData: {
+                key: ONYXKEYS.IS_LOADING_REPORT_DATA,
+            },
+            betas: {
+                key: ONYXKEYS.BETAS,
+            },
+            policies: {
+                key: ONYXKEYS.COLLECTION.POLICY,
+            },
+            reportActions: {
+                key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${route.params.reportID}`,
```

Comment:
Original code explicitly added `toString()`. Would this break something if we don't put `toString`?